### PR TITLE
Authentication and QOL changes

### LINKED ISSUE #6: Support authentication
Currently the Xpra application doesn't support any mode of authentication, at least username/password for PAM authentication would suffice, if you wouldn't mind adding this feature to the client!

### PR DESCRIPTION
**Pull request by ShadowEO**
- Added authentication as per my request #6
- Moved Settings dialog into it's own menu option to help facilitate saving settings.
- Added rudimentary auto-login feature.

Currently the password is saved in plain-text in the client storage (whether this be on the server, or in the browser). This at least satisfies the use-case of authenticated Xpra use, if someone can provide a more secure way to store/retrieve the password, I'd be open to it.

(Recreated on request)

### CODE REVIEW

**Comment by ShadowEO** [comment]
Hopefully this PR works better for you. Great project btw, combining it with Xpra makes it exceptionally powerful. You may be seeing more contributions from me in the future as I attempt to make OS.js suit my needs!

**Comment by andersevenrud** [comment]
Thanks!

And it seems like the workflow triggered this time :tada:

**Comment by ShadowEO** [comment]
I see some linting issues, I'll work to fix those when I get back to my workstation

**Comment by andersevenrud** [comment]
Yup. I'll finish up the review once you get those fixed :smile:

**Comment by ShadowEO** [comment]
Alright, I fixed everything eslint complained about though there was one line that threw a warning (which was part of the original code, I didn't touch the websocket parts).

File: index.js [hunk 1]
```diff
@@ -38,10 +38,15 @@ const register = (core, args, options, metadata) => {
   let tray;
   let status = 'disconnected';
   let windows = [];
-
-  const defaultOptions = {
-    uri: 'ws://localhost:10000',
-    sound: false
+  const proc = core.make('osjs/application', {args, options, metadata});
+  let defaultOptions = {
```
Comment (andersevenrud) [inline index.js]: Revert this back to a `const`.

File: index.js [hunk 2]
```diff
@@ -108,14 +149,27 @@ const register = (core, args, options, metadata) => {
...
             ? client.disconnect()
-            : createConnectionDialog(options => client.connect(options))
+            : client.connect(defaultOptions)
+        }, {
+          label: 'Settings',
+          onclick: () =>
+            createConnectionDialog(options => {
+              proc.settings = options;
+              proc.saveSettings();
+              defaultOptions = options;
```
Comment (andersevenrud) [inline index.js]: Then change this to `Object.assign(defaultOptions, options)`

**Review by andersevenrud — CHANGES_REQUESTED** [review]

**Review by ShadowEO — COMMENTED** [review]
I've made the changes in my fork as requested. Please review at your leisure :D

**Review by andersevenrud — APPROVED** [review]
LGTM. I still need to take it for a quick test drive before merging :)

**Pull request merged**